### PR TITLE
feat: Update docker container scanning flag

### LINKED ISSUE #1316: Make `--docker` flag use new container scanning
Do `docker pull` and `docker save` under the hood for convenience.

### PR DESCRIPTION
**Pull request by another-rex**
Resolves #1316 

Update the --docker flag to:
- Only accept one image to scan at a time (to make displaying results easier)
- Call new image scanning function internally.
- Acts like a convenience function for 
```
docker save <image-name> > img-name.tar && osv-scanner --experimental-oci-image=img.name.tar
```

TODO: 
- [x] Add an ACCEPTANCE test which uses docker to pull down a stable image. 
- [x] Include a docker pull first, as docker save only saves images already on device and does not pull images online.

### CODE REVIEW

File: pkg/osvscanner/osvscanner.go [hunk 1]
```diff
@@ -644,72 +644,78 @@ func createCommitQueryPackage(commit string, source string) scannedPackage {
...
+	err = tempImageFile.Close()
 	if err != nil {
-		r.Errorf("Failed to get stderr: %s\n", err)
 		return nil, err
 	}
+	defer os.Remove(tempImageFile.Name())
 
-	err = cmd.Start()
+	r.Infof("Pulling docker image (%q)...\n", dockerImageName)
+	err = runCommandLogError(r, "docker", "pull", "-q", dockerImageName)
```
Comment (oliverchang) [inline pkg/osvscanner/osvscanner.go:696]: What's the CLI output if `docker` doesn't exist on the user's system?

Comment (another-rex) [reply pkg/osvscanner/osvscanner.go:696]:
```
            Failed to run docker command ("docker pull -q alpine:non-existent-tag"): exec: "docker": executable file not found in $PATH
            exec: "docker": executable file not found in $PATH
```

File: pkg/osvscanner/osvscanner.go [hunk 2]
```diff
@@ -644,72 +644,78 @@ func createCommitQueryPackage(commit string, source string) scannedPackage {
...
-			Version:   splitText[1],
-			Ecosystem: "Debian",
-			Source: models.SourceInfo{
-				Path: dockerImageName,
-				Type: "docker",
-			},
-		})
+	// Modify the image path to be the image name, rather than the temporary file name
+	for i := range packages {
+		pathToModify := &packages[i].Source.Path
```
[the remark this answers is not in the document]
Comment (another-rex) [reply pkg/osvscanner/osvscanner.go]: No particular reason, updated!

**Review by hogo6002 — APPROVED** [review]

File: pkg/osvscanner/osvscanner.go (hunk 1 above)
Comment (oliverchang) [reply pkg/osvscanner/osvscanner.go:696]: nice! that's pretty clear to the user.

**Pull request merged**